sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -372,14 +390,57 @@ public BoundedWindow window() {
   public @interface StartBundle {}
 
   /**
-   * Annotation for the method to use for processing elements. A subclass of
-   * {@link DoFn} must have a method with this annotation satisfying
-   * the following constraints in order for it to be executable:
+   * Annotation for the method to use for processing elements. A subclass of {@link DoFn} must have
+   * a method with this annotation.
+   *
+   * <p>The signature of this method must satisfy the following constraints:
+   *
    * <ul>
-   *   <li>It must have at least one argument.
-   *   <li>Its first argument must be a {@link DoFn.ProcessContext}.
-   *   <li>Its remaining argument, if any, must be {@link BoundedWindow}.
+   * <li>Its first argument must be a {@link DoFn.ProcessContext}.
+   * <li>Its remaining arguments, if any, must be either {@link BoundedWindow} or a subtype of
```

Comment:
Thinking about maintaining this moving forward, I wonder if it would be easier to have something like:

"If it takes a {@link RestrictionTracker} as an argument, then it is a <a href...>splittable</a> DoFn and is subject to the requirements specified {@link below section}. Otherwise, the following restrictions apply:"

That way we don't have to merge all SDF parameter notes into this, and maintain them seprately (eg., this sentence seems like like will become difficult as we add more arguments to both, etc.)